doc: Add log configuration page CY-2768 (#713)
codacy/chart: docs/configuration/logging.md

---

> description: Instructions on how to configure log levels and log retention periods for your Codacy Self-hosted instance.

Reviewer: The meta description can be made slightly shorter and more assertive:

```suggestion
description: Configure log levels and log retention periods for your Codacy Self-hosted instance.
```

---

# Logging

> Currently, Codacy Self-hosted supports two types of log configuration:

Reviewer: ```suggestion
Currently, Codacy Self-hosted supports two types of log configurations:
```

> -   **[Log level](#setting-up-log-levels):** The severity of the messages that will be shown in logs (we recommend not going lower than WARN, as the logs are used to troubleshoot any problems).

Reviewer: I propose some style changes to the sentence:

```suggestion
-   **[Log level](#setting-up-log-levels):** The severity of the messages that will be shown in logs. We recommend that you don't use a log level lower than WARN, as the logs are useful to troubleshoot any issues in your Codacy Self-hosted instance.
```

-   **[Log retention period](#configuring-log-retention-period):** The period during which logs will be stored, before being removed by the cleanup job specified in the configuration.

The sections below provide instructions on how to configure each logging configuration.

## Setting up log level

The log level of most components can be configured by editing the value of `COMPONENT.config.logLevel` in the `values-production.yaml` file that is used to install Codacy, like is shown in the example below:
```
worker-manager:
  replicaCount: 2
  config:
    logLevel: WARN
  resources:
    limits:
      cpu: 500m
      memory: 1000Mi
    requests:
      cpu: 200m
      memory: 200Mi
```

> The components that currently support this configuration are `codacy-api`, `listener`, `engine`, and `worker-manager`.
> And the list of supported values for this configuration are shown below. Note that each level also prints the information of the levels higher than it:
> * DEBUG - **Most** verbose level. Logs detailed information on the flow through the system.
> * INFO - Logs interesting events such as application startup and shutdown, important events on the flow of the system, or behaviour that is skipped for expected reasons.
> * WARN - Logs runtime events that are unexpected, or undesirable, but not necessarily wrong such as use of deprecated APIs.
> * ERROR - **Least** verbose level. Only logs runtime errors, or unexpected conditions that prevent the expected behaviour from happening.

Reviewer: Some small edits and formatting suggestions (such as using `-   ` for list items, the [convention used in the documentation](https://github.com/codacy/docs/blob/master/CONTRIBUTING.md#markdown-conventions)):

```suggestion
The components that currently support this configuration are `codacy-api`, `listener`, `engine`, and `worker-manager`.

The list of supported values for this configuration is shown below. Note that each level also prints the information of the levels higher than it:

-   DEBUG: **Most** verbose level. Logs detailed information on the flow through the system.
-   INFO: Logs interesting events such as application startup and shutdown, important events on the flow of the system, or behavior that is skipped for expected reasons.
-  WARN: Logs runtime events that are unexpected, or undesirable, but not necessarily wrong such as the use of deprecated APIs.
-  ERROR: **Least** verbose level. Only logs runtime errors, or unexpected conditions that prevent the expected behavior from happening.
```

Author: thank you!

---
description: Configure log levels and log retention periods for your Codacy Self-hosted instance.
---

# Logging

Currently, Codacy Self-hosted supports two types of log configurations:

> -   **[Log level](#setting-up-log-levels):** The severity of the messages that will be shown in logs. We recommend that you don't use a log level lower than WARN, as the logs are useful to troubleshoot any issues in your Codacy Self-hosted instance.
> -   **[Log retention period](#configuring-log-retention-period):** The period during which logs will be stored, before being removed by the cleanup job specified in the configuration.

Reviewer: Updates the links to the section titles:

```suggestion
-   **[Log level](#configuring-the-log-level):** The severity of the messages that will be shown in logs. We recommend that you don't use a log level lower than WARN, as the logs are useful to troubleshoot any issues in your Codacy Self-hosted instance.
-   **[Log retention period](#configuring-the-log-retention-period):** The period during which logs will be stored, before being removed by the cleanup job specified in the configuration.
```

The sections below provide instructions on how to configure each logging configuration.

> ## Setting up log level

Reviewer: Suggestion to make the two section titles consistent:

```suggestion
## Configuring the log level
```

The log level of most components can be configured by editing the value of `COMPONENT.config.logLevel` in the `values-production.yaml` file that is used to install Codacy, like is shown in the example below:
```yaml
worker-manager:
  replicaCount: 2
  config:
    logLevel: WARN
  resources:
    limits:
      cpu: 500m
      memory: 1000Mi
    requests:
      cpu: 200m
      memory: 200Mi
```
The components that currently support this configuration are `codacy-api`, `listener`, `engine`, and `worker-manager`.

The list of supported values for this configuration is shown below. Note that each level also prints the information of the levels higher than it:

> -   DEBUG: **Most** verbose level. Logs detailed information on the flow through the system.
> -   INFO: Logs interesting events such as application startup and shutdown, important events on the flow of the system, or behavior that is skipped for expected reasons.
> -  WARN: Logs runtime events that are unexpected, or undesirable, but not necessarily wrong such as the use of deprecated APIs.
> -  ERROR: **Least** verbose level. Only logs runtime errors, or unexpected conditions that prevent the expected behavior from happening.

Reviewer: There were a couple of missing spaces in my previous suggestion, sorry! :sweat_smile: 

```suggestion
-   DEBUG: **Most** verbose level. Logs detailed information on the flow through the system.
-   INFO: Logs interesting events such as application startup and shutdown, important events on the flow of the system, or behavior that is skipped for expected reasons.
-   WARN: Logs runtime events that are unexpected, or undesirable, but not necessarily wrong such as the use of deprecated APIs.
-   ERROR: **Least** verbose level. Only logs runtime errors, or unexpected conditions that prevent the expected behavior from happening.
```

Reviewer 2: @prcr @lventura-codacy My only suggestion is that we let the reader know what the default log level is, if possible. Is the default the same for each component?

Reviewer: @dan-codacy, we now mention that DEBUG is the default log level (it's the same for all components).

Could you please have another look at the page? You can see the Markdown preview if that helps:

https://github.com/codacy/chart/blob/log-configuration-doc/docs/configuration/logging.md

> ## Configuring log retention period

Reviewer: It's more correct to include the article "the" because we're referring to a specific "log retention period" ([here are some rules](https://www.englishgrammar.org/omission-articles/) for when we can leave the article out if you're curious :nerd_face:):

```suggestion
## Configuring the log retention period
```

> As stated in the `values.yaml` file used during deployments, if the `fluentdoperator` is enabled (`enabled:true`), the logs will be collected, and must be cleaned.

Reviewer: We can simplify the way we refer to a setting by using the "full path" of the configuration key:

```suggestion
As stated in the `values.yaml` file used during deployments, if `fluentdoperator.enabled: true` the logs will be collected and must be cleaned.
```

> To cleanup a bucket in minio, it is necessary to create a bucket lifecycle policy. In the chart repository, Codacy provides a cleaning job template ([lifecycle-police-job.yaml](https://github.com/codacy/chart/blob/master/codacy/templates/fluentd/lifecycle-police-job.yaml)), which will create the policy for you when included in your deployment.

Reviewer: We need to ensure that we link to a version of the `yaml` configuration file that is synchronized with the Codacy Self-hosted and documentation version, and not to the one on the `master` branch. Please see the solution adopted in https://github.com/codacy/chart/pull/515, but let me know if you need any help. :+1:

Reviewer: Now that we saw that customers don't actually need to download and edit this file, it's OK to have a direct link to the version in `master` because it's only meant as a way for customers to see what's included in the file. :+1:
